vignettes/join-keys.Rmd

```diff
@@ -10,90 +12,256 @@ vignette: >
+| `##   --> ds1: [col_1]`          |                                                                               | 
+| `##   --* (implicit via parent with): ds2` |                                                                               | 
+| `## ds4: [no primary keys]`    |                                                                               | 
+| `##   <-> ds5: [col_5]`          |  **Foreign keys** <br />_(arrow `<->` denotes no parent definition between datasets)_      | 
+| `## ds5: [no primary keys]`    |                                                                               | 
+| `##   <-> ds4: [col_4] `         |                                                                               | 
+
+## Accessing and Modifying keys
+
+The `join_keys` object behaves like a list and the preferred method to get a specific key to use the _subset_ operator with 2 indices. 
+
+```{r}
+# Using the jk object defined on "Anatomy of Join Keys"
+jk
+
+# Getting primary key of "ds1"
+jk["ds1", "ds1"]
+
+# Getting foreign keys between "ds2" and "ds1"
+jk["ds4", "ds5"]
+
+# Note the symmetry on the keys
+jk["ds5", "ds4"]
+```
+
+Modifying or adding a key uses the same notation with the assignment operator `<-`. This will ensure symmetry as it will create both relationships. Assigning `NULL` value will delete the relation.
+
+```{r}
+# Adding a new ds5 <-> ds1 key
+jk["ds5", "ds1"] <- "a_column"
+
+# Removing an existing key
+jk["ds4", "ds5"] <- NULL
+```
+
+## Join Keys Relationships
+
+There are 2 types of relationships encoded with `join_keys` described in the following sections. The _primary_ and _foreign_ keys are created explicitly using the `join_key` function.
+
+Additionally, the `join_keys` object detects implicit relationships when two datasets share foreign keys to a parent dataset, but not between themselves. These implicit relations are available just like another foreign key and can be used to merge datasets, despite not being defined by the user.
+
+### Primary Key with _teal_data_
+
+When using the `teal_data` function, the simplest method to define the join keys  is to use the `join_keys` argument.
+We can specify the column(s) of the dataset that _(together)_ uniquely identify rows in the dataset.
+
+```{r, include=FALSE}
+# nolint start: commented_code_linter.
+```
 
 ```{r, message=FALSE}
 library(teal.data)
 library(dplyr)
 
-iris_with_keys <- mutate(iris, id = factor(row_number()))
-
-ds <- dataset(
-  "IRIS",
-  iris_with_keys,
-  code = "IRIS <- iris %>% mutate(id = factor(row_number()))",
-  keys = c("id")
+td_pk <- within(
+  teal_data(),
+  ds1 <- mutate(iris, id = factor(row_number()))
 )
+datanames(td_pk) <- c("ds1")
+
+join_keys(td_pk) <- join_keys(join_key("ds1", keys = "id"))
+## Alternatively, when only a single primary key is defined the `join_keys()`
+## the wrapper can be skipped
+# join_keys(td_pk) <- join_key("ds1", keys = "id")
```

Comment:
I wouldn't necessarily write about this. Let's promote one way and don't give too much informations (alternatives).

```suggestion
```